<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>EquitWebServer::MainWindow</class>
 <widget class="EquitWebServer::Window" name="EquitWebServer::MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>940</width>
    <height>491</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout" rowstretch="1,0" columnstretch="0,1">
    <item row="0" column="0">
     <widget class="QLabel" name="logo">
      <property name="pixmap">
       <pixmap resource="../resources/resources.qrc">:/logo/app</pixmap>
      </property>
      <property name="alignment">
       <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
      </property>
      <property name="margin">
       <number>2</number>
      </property>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="EquitWebServer::ConfigurationWidget" name="configuration" native="true"/>
    </item>
    <item row="1" column="1">
     <layout class="QHBoxLayout" name="buttonsLayout">
      <item>
       <widget class="QPushButton" name="saveDefault">
        <property name="toolTip">
         <string>Save the current settings as the default configuration.</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset theme="document-save">
          <normaloff>.</normaloff>.</iconset>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="startStop">
        <property name="text">
         <string>Start</string>
        </property>
        <property name="icon">
         <iconset theme="media-playback-start" resource="../resources/resources.qrc">
          <normaloff>:/icons/buttons/startserver</normaloff>:/icons/buttons/startserver</iconset>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="quit">
        <property name="text">
         <string>Quit</string>
        </property>
        <property name="icon">
         <iconset theme="application-exit" resource="../resources/resources.qrc">
          <normaloff>:/icons/buttons/exit</normaloff>:/icons/buttons/exit</iconset>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>940</width>
     <height>23</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuServer">
    <property name="title">
     <string>Server</string>
    </property>
    <addaction name="actionOpenConfiguration"/>
    <addaction name="actionSaveConfiguration"/>
    <addaction name="actionSaveDefaultConfiguration"/>
    <addaction name="separator"/>
    <addaction name="actionRecentConfigurations"/>
    <addaction name="separator"/>
    <addaction name="actionDocumentRoot"/>
    <addaction name="actionListenOnLocalhost"/>
    <addaction name="actionListenOnHostAddress"/>
    <addaction name="separator"/>
    <addaction name="actionStart"/>
    <addaction name="actionStop"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuAccess">
    <property name="title">
     <string>Access</string>
    </property>
    <addaction name="actionAllowUnknownIps"/>
    <addaction name="actionForbidUnknownIps"/>
    <addaction name="actionClearIpPolicyList"/>
   </widget>
   <widget class="QMenu" name="menuContent">
    <property name="title">
     <string>Content</string>
    </property>
    <addaction name="actionClearAllMimeAssociations"/>
    <addaction name="actionClearAllMimeActions"/>
    <addaction name="separator"/>
    <addaction name="actionResetDefaultMimeType"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <addaction name="menuServer"/>
   <addaction name="menuAccess"/>
   <addaction name="menuContent"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="EquitWebServer::MainWindowStatusBar" name="statusbar"/>
  <action name="actionOpenConfiguration">
   <property name="icon">
    <iconset theme="document-open" resource="../resources/resources.qrc">
     <normaloff>:/icons/menu/openconfig</normaloff>:/icons/menu/openconfig</iconset>
   </property>
   <property name="text">
    <string>Open configuration...</string>
   </property>
   <property name="toolTip">
    <string>Open a saved configuration.</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSaveConfiguration">
   <property name="icon">
    <iconset theme="document-save" resource="../resources/resources.qrc">
     <normaloff>:/icons/menu/saveconfig</normaloff>:/icons/menu/saveconfig</iconset>
   </property>
   <property name="text">
    <string>Save configuration...</string>
   </property>
   <property name="toolTip">
    <string>Save the current configuration.</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionDocumentRoot">
   <property name="icon">
    <iconset theme="folder-open" resource="../resources/resources.qrc">
     <normaloff>:/icons/menu/choosedocumentroot</normaloff>:/icons/menu/choosedocumentroot</iconset>
   </property>
   <property name="text">
    <string>Document root...</string>
   </property>
   <property name="toolTip">
    <string>Choose the document root for the server.</string>
   </property>
  </action>
  <action name="actionListenOnLocalhost">
   <property name="text">
    <string>Listen on localhost</string>
   </property>
   <property name="toolTip">
    <string>Set the server to listen on the localhost loopback address.</string>
   </property>
  </action>
  <action name="actionListenOnHostAddress">
   <property name="text">
    <string>Listen on host address</string>
   </property>
   <property name="toolTip">
    <string>Set the server to listen on the first IP address assigned to this host.</string>
   </property>
  </action>
  <action name="actionStart">
   <property name="icon">
    <iconset theme="media-playback-start" resource="../resources/resources.qrc">
     <normaloff>:/icons/menu/startserver</normaloff>:/icons/menu/startserver</iconset>
   </property>
   <property name="text">
    <string>Start</string>
   </property>
   <property name="toolTip">
    <string>Start the web server.</string>
   </property>
  </action>
  <action name="actionStop">
   <property name="icon">
    <iconset theme="media-playback-stop" resource="../resources/resources.qrc">
     <normaloff>:/icons/menu/stopserver</normaloff>:/icons/menu/stopserver</iconset>
   </property>
   <property name="text">
    <string>Stop</string>
   </property>
   <property name="toolTip">
    <string>Stop the web server.</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="icon">
    <iconset theme="application-exit" resource="../resources/resources.qrc">
     <normaloff>:/icons/buttons/exit</normaloff>:/icons/buttons/exit</iconset>
   </property>
   <property name="text">
    <string>Quit</string>
   </property>
   <property name="toolTip">
    <string>Quit the application.</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionSaveDefaultConfiguration">
   <property name="text">
    <string>Save default configuration...</string>
   </property>
   <property name="toolTip">
    <string>Save the current configuration as the default configuration for the server.</string>
   </property>
  </action>
  <action name="actionRecentConfigurations">
   <property name="text">
    <string>Recent configurations</string>
   </property>
   <property name="toolTip">
    <string>Configurations that have been used recently.</string>
   </property>
  </action>
  <action name="actionAllowUnknownIps">
   <property name="icon">
    <iconset theme="dialog-ok" resource="../resources/resources.qrc">
     <normaloff>:/icons/connectionpolicies/accept</normaloff>:/icons/connectionpolicies/accept</iconset>
   </property>
   <property name="text">
    <string>Allow unknown IPs</string>
   </property>
   <property name="toolTip">
    <string>Allow connections from IP addresses for which no specific policy is set.</string>
   </property>
  </action>
  <action name="actionForbidUnknownIps">
   <property name="icon">
    <iconset theme="dialog-cancel" resource="../resources/resources.qrc">
     <normaloff>:/icons/connectionpolicies/reject</normaloff>:/icons/connectionpolicies/reject</iconset>
   </property>
   <property name="text">
    <string>Forbid unknown IPs</string>
   </property>
   <property name="toolTip">
    <string>Forbid connections from IP addresses for which no specific policy is set.</string>
   </property>
  </action>
  <action name="actionClearIpPolicyList">
   <property name="text">
    <string>Clear IP policy list</string>
   </property>
   <property name="toolTip">
    <string>&lt;p&gt;Clear the list of policies for specific IP addresses.&lt;/p&gt;
&lt;p&gt;&lt;small&gt;All connections will subsequently use the default connection policy.&lt;/small&gt;&lt;/p&gt;</string>
   </property>
  </action>
  <action name="actionClearAllMimeAssociations">
   <property name="text">
    <string>Clear all MIME type associations</string>
   </property>
   <property name="toolTip">
    <string>&lt;/p&gt;Remove all associations between file extensions and MIME types.&lt;/p&gt;
&lt;p&gt;&lt;small&gt;Subsequent requests will treat all files as having the default MIME type.&lt;/small&gt;&lt;/p&gt;</string>
   </property>
  </action>
  <action name="actionClearAllMimeActions">
   <property name="text">
    <string>Clear all MIME type actions</string>
   </property>
   <property name="toolTip">
    <string>&lt;p&gt;Clear all actions registered for specific MIME types.&lt;/p&gt;
&lt;p&gt;&lt;small&gt;Subsequent requests will execute the default action for all MIME types.&lt;/small&gt;&lt;/p&gt;</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="icon">
    <iconset theme="help-about" resource="../resources/resources.qrc">
     <normaloff>:/icons/menu/about</normaloff>:/icons/menu/about</iconset>
   </property>
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="actionResetDefaultMimeType">
   <property name="text">
    <string>Reset default MIME type</string>
   </property>
   <property name="toolTip">
    <string>&lt;p&gt;Reset the default mime type to &lt;em&gt;application/octet-stream&lt;/em&gt;.&lt;/p&gt;</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>EquitWebServer::Window</class>
   <extends>QMainWindow</extends>
   <header>src/window.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>EquitWebServer::ConfigurationWidget</class>
   <extends>QWidget</extends>
   <header>src/configurationwidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>EquitWebServer::MainWindowStatusBar</class>
   <extends>QStatusBar</extends>
   <header>src/mainwindowstatusbar.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../resources/resources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>saveDefault</sender>
   <signal>clicked()</signal>
   <receiver>actionSaveDefaultConfiguration</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>21</x>
     <y>445</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
